rpc/__tests__/notary.ts

```diff
@@ -0,0 +1,426 @@
+import { uuidv4 } from "../../src/utils";
+import { Wallet } from "../../src/entity/Wallet";
+import { EncryptorContainer } from '../../src/entity/encryption/EncryptorContainer';
+import { latest as LATEST_NOTARY } from '../VaultNotary/Latest';
+
+export const RPCVaultNotaryTests = async function() {
+
+    let fullWallet1;
+    let fullWallet2;
+
+    beforeAll(async () => {
+        await EncryptorContainer.init();
+        // Import known funded wallets
+        fullWallet1 = await Wallet.import_entity('0x0000000000000000000000000000000000000000000000000000000000000001');
+        await fullWallet1.save();
+        fullWallet2 = await Wallet.import_entity('0x0000000000000000000000000000000000000000000000000000000000000002');
+        await fullWallet2.save();
+    });
+
+    afterAll(async() => {
+        await cleanupEntities(typeorm);
+    });
+
+    describe('RegisterVault', function() {
+        it(`Has required parameters`, mochaAsync(async () => {
+            let caughtError;
+
+            try {
+                await CallRPCMethod(RPCVaultNotary.RegisterVaultTx, {});
+                fail("Did not Throw"); return;
+            } catch (err) {
+                caughtError = err;
+            }
+
+            expectMissingRequiredParams(caughtError, ['vaultId', 'senderWallet', 'vaultUri', 'vaultHash']);
+        }));
+
+        it(`Validates UUID parameters`, mochaAsync(async () => {
+            let caughtError;
+
+            try {
+                await CallRPCMethod(RPCVaultNotary.RegisterVaultTx, {
+                    senderWallet: '123',
+                    vaultId: '123',
+                    vaultUri: '123',
+                    vaultHash: '123',
+                });
+                fail("Did not Throw"); return;
+            } catch (err) {
+                caughtError = err;
+            }
+
+            expectInvalidUUIDParams(caughtError, ['vaultId', 'senderWallet']);
+        }));
+
+
+        it(`Validate properties exist in the build transaction return`, mochaAsync(async () => {
+                let rpcReturn = await CallRPCMethod(RPCVaultNotary.RegisterVaultTx, {
+                    vaultId: uuidv4(),
+                    senderWallet: fullWallet1.id,
+                    vaultUri: '123',
+                    vaultHash: '123'
+                });
+
+                expect(rpcReturn).toBeDefined();
+                expect(rpcReturn.success).toBeTruthy();
+                expect(rpcReturn.transaction.nonce).toBeDefined();
+                expect(rpcReturn.transaction.gasPrice).toBeDefined();
+                expect(rpcReturn.transaction.gasLimit).toBeDefined();
+                expect(rpcReturn.transaction.value).toBeDefined();
+                expect(rpcReturn.transaction.data).toBeDefined();
+                expect(rpcReturn.transaction.to).toBeDefined();
+                expect(rpcReturn.transaction.chainId).toBeDefined();
+                expect(rpcReturn.contractVersion).toMatch(LATEST_NOTARY);
+ 
+        }));
+
```

Comment:
If you validate the vaultUri and vaultHash RPC parameters as Strings, put some tests in here where you provide non-string values.  There should be example tests that utilize the `expectInvalidStringParams` test assertion helper.

Reply:
This test assertion helper is very handy. Thanks!